scripts/o3de/o3de/repo_properties.py

```diff
@@ -484,6 +647,9 @@ def add_parser_args(parser):
     modify_object_group.add_argument('--download-prefix', '-dp', type=str, required=False,
                                    help='A URL prefix for a file attached to a GitHub release might look like this:'
                                         '-dp https://github.com/o3de/o3de-extras/releases/download/2305.0/')
+    modify_object_group.add_argument('--upload-github-release-tag', '-ugrt', type=str, default=False,
```

Comment:
I think it'd be more generic and future proof to rename everywhere we use `upload-github-release-tag` to `upload-git-release-tag` so we can add other Git providers and not need to change this parameter when we do, we can usually determine if it's GitHub or not by looking at the URL which is what we do for download URLs.